os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -580,11 +580,8 @@ static inline void print_assert_detail(const uint8_t *filename, int lineno, stru
 
 void up_assert(const uint8_t *filename, int lineno)
 {
-	/* ARCH_GET_RET_ADDRESS should always be
-	 * called at the start of the function */
 
-	size_t kernel_assert_location = 0;
-	ARCH_GET_RET_ADDRESS(kernel_assert_location);
+	size_t kernel_assert_location = (size_t)__builtin_return_address(0);
```

Comment:
When we use the `ARCH_GET_RET_ADDRESS`, we should get the address at start of the function. That's why we define the local variable - kernel_assert_location and use it line 611 only.
But with the new function, we don't need to save it here. I mean no need to use the variable in this purpose.